### PR TITLE
Common/Adv. Fatigue/Medical - Add unified sway calculation

### LINKED ISSUE #8407: Advanced Fatigue - Medical overrides `swayFactor` setting
**Mods:**
- **Arma 3:** `2.04.147716`
- **CBA:** `3.15.5.210713`
- **ACE3:** `3.13.6` (https://github.com/acemod/ACE3/commit/8bd4922fccc86b53bae25d43a8c95b77c9b75f54)

**Description:**
Setting Adv. Fatigue sway factor lower than 1 gets overridden by hardcoded Medical sway factor

**Steps to reproduce:**

1. Enable Adv. Fatigue.
2. Set `ace_advanced_fatigue_swayFactor` to anything lower than `1`, `0.5` in this case.
3. Inspect the returns from the sway calcs:

```sqf
private _aimHash = (missionNamespace getVariable ["ACE_setCustomAimCoef", nil]);

{
    private _hashCode = [_aimHash, _x] call CBA_fnc_hashGet;

    diag_log text format ["%1 sway value: %2", _x, call _hashCode];
} forEach ([_aimHash] call CBA_fnc_hashKeys);

private _swayMax = [missionNamespace, "ACE_setCustomAimCoef", "max"] call ace_common_fnc_arithmeticGetResult;

diag_log text format ["Sway value from arithmeticGetResult: %1", _swayMax];
```

```
21:47:53 ace_advanced_fatigue sway value: 0.782006
21:47:53 ace_medical sway value: 1
21:47:53 Sway value from arithmeticGetResult: 1
```

**Expected behavior:**
Sway/amingCoef affected by `ace_advanced_fatigue_swayFactor` setting.

**Where did the issue occur?**
Anywhere

**Additional context:**
https://github.com/acemod/ACE3/blob/76676eee462cb0bbe400a482561c148d8652b550/addons/advanced_fatigue/XEH_postInit.sqf#L5-L20

https://github.com/acemod/ACE3/blob/76676eee462cb0bbe400a482561c148d8652b550/addons/medical/XEH_postInit.sqf#L8-L10

Overwriting the `ace_medical` `ACE_setCustomAimCoef` codeblock with `ace_advanced_fatigue_swayFactor` incorporated solves the issue, don't know if it's the best solution though:

```sqf
[missionNamespace, "ACE_setCustomAimCoef", "ace_medical", {
    ((linearConversion [0, 1, (0 max ((ACE_player getVariable ["ace_medical_pain", 0]) - (ACE_player getVariable ["ace_medical_painSuppress", 0])) min 1), 1, 5, true]) + (ACE_player getVariable ["ace_medical_engine_aimFracture", 0])) * ace_advanced_fatigue_swayFactor
}] call ace_common_fnc_arithmeticSetSource;
```

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Move sway calculation and application to Common, make Adv. Fatigue and Medical use it
- Add API for setting sway baselines and multipliers. Highest baseline will be used, then multiplied by the product of all multipliers.
- Close #8407

See #8486 and #8407 for reference.

I'm happy with this system but numbers will need a bit of tweaking. @BaerMitUmlaut

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by LinkIsGrim** [comment]
On a side-note: should crouch's multiplier really be lower than prone? The end result is as expected and you end up getting less sway in prone still, but that looks weird.

**Comment by BaerMitUmlaut** [comment]
Yes, there is a default value applied by vanilla regardless of factor, which is why the factor needs to be lower.

**Comment by LinkIsGrim** [comment]
Given how often this has been attempted, not coming to a consensus on how it should work (mechanic and value-wise), and repeated mentions of moving the actual use of the engine command to common, I'll be marking this as draft until we work out a proper solution, starting by moving to common since we can all agree on that at least.

**Comment by Whigital** [comment]
Close https://github.com/acemod/ACE3/pull/9140, since it's way too overcomplicated compared to this solution ?

**Comment by LinkIsGrim** [comment]
> Close #8486, since it's way too overcomplicated compared to this solution ?

No harm in keeping it open for now.

**Review by PabstMirror — APPROVED** [review]

File: addons/common/functions/fnc_addSwayFactor.sqf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+*/
+params ["_type", "_code", "_id"];
+
+_type = toLower _type;
+
+if !(_type in ["baseline", "multiplier"]) exitWith { ERROR_2("%1-%2 type unsupported",_type,_id); false };
+
+if !((call _code) isEqualType 0) exitWith { ERROR_2("%1-%2 bad return type",_type,_id); false };
+
+[missionNamespace, format ["ACE_setCustomAimCoef_%1", toLower _type], _id, _code] call FUNC(arithmeticSetSource);
```
Comment (LinkIsGrim) [inline addons/common/functions/fnc_addSwayFactor.sqf:27]:
```suggestion
[missionNamespace, format ["ACE_setCustomAimCoef_%1", _type], _id, _code] call FUNC(arithmeticSetSource);
```

Redundant toLower

**Pull request merged**